src/gen/types.rs

```diff
@@ -136,14 +137,19 @@ impl TypeHelperRenderer for TypeHelpersRenderer<'_> {
 }
 
 impl Renderer<(FunctionDefinition, dart::Tokens)> for TypeHelpersRenderer<'_> {
+    // TODO: Implimient a two pass system where the first pass will render the main code, and the second pass will render the helper code
```

Comment:
please do not put FIXME or TODOs in the code. They belong as issues in the issue tracker.